pkg/adapter/check.go

```diff
@@ -31,3 +31,24 @@ type CheckResult struct {
 	// ValidUseCount represents the number of uses for which this result can be considered valid.
 	ValidUseCount int64
 }
+
+// GetStatus gets status embedded in the result.
+func (r *CheckResult) GetStatus() rpc.Status { return r.Status }
+
+// SetStatus embeds status in result.
+func (r *CheckResult) SetStatus(s rpc.Status) { r.Status = s }
+
+// Combine combines other result with self.
+func (r *CheckResult) Combine(otherPtr interface{}) interface{} {
```

Comment:
This doesn't do anything with Status? Is this on purpose? A comment to that effect would be good.